### PR TITLE
Update frontend tests

### PR DESCRIPTION
**Pull request by nspool**
Fix list screens so they will render correctly without data and not stay in a loading state.
Add/Update react-testing-library, jest and msw for mocking API calls.
Start creating new frontend tests starting with LoginForm.

Checkout this branch and use `yarn test` to verify it is working. Alternatively the Actions tab should show the CI result.

### CODE REVIEW

**Comment by utas-raymondng** [comment]
I see that animal-tracking use cypress as the test framework for Javascript but here we do not use it. Wonder if we have some standard practice on what to use for testing across projects?

**Review by utas-raymondng — CHANGES_REQUESTED** [review]

**Comment by gsatimos** [comment]
> I see that animal-tracking use cypress as the test framework for Javascript but here we do not use it. Wonder if we have some standard practice on what to use for testing across projects?

I've got a PR up to remove some residual cypress files and folders from animal-tracking UI (https://github.com/aodn/animal-tracking/pull/1550 - although build is failing so I need to fix that). 

So yeah, the animal tracking API uses cypress to run integration tests and has good coverage whereas NRMN doesn't use cypress and has tests written in Java. The guy who kicked off most of the development for NRMN took many things in a different direction from animal tracking which @nspool has since refactored. That said, I think we missed an opportunity to align things at the point where we were doing some new application development.

In a nutshell, no - no standard practice

File: web/src/api/api.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ axiosInstance.interceptors.response.use(
     return response;
   },
   (error) => {
-    window.setApplicationError(error?.message || JSON.stringify(error), error);
+    window.setApplicationError && window.setApplicationError(error?.message || JSON.stringify(error), error);
```
[the remark this answers is not in the document]
Comment (nspool) [reply web/src/api/api.js:20]: This function will always be defined except for the Jest tests. Because of this I think it's OK to do it this way.

I agree if this was a third-party library, something where the method maybe a different type, or this was being exposed as a API then it would be better to do a type check.

[the remark this answers is not in the document]
Comment (utas-raymondng) [reply web/src/api/api.js:20]: That is fine, but can you add a comment there so that new comer knows why do it this way.

[the remark this answers is not in the document]
Comment (nspool) [reply web/src/api/api.js]: Change the code to remove the 'truthy' test and instead add an empty setApplicationError method if it's not present on the window.

**Review by utas-raymondng — APPROVED** [review]

**Comment by nspool** [comment]
Created `test-failure-test` to verify that broken tests so as failed in Jenkins and GitHub.
Confirmation: https://github.com/aodn/nrmn-application/actions/runs/2061433168
